invenio_oauthclient/handlers/base.py

```diff
@@ -46,14 +47,63 @@
 )
 
 
+def _role_needs_update(role_obj, new_role_dict):
+    """Checks if role needs to be updated."""
+    if role_obj.name != new_role_dict.get("name") or role_obj.description != new_role_dict.get("description"):
+        return True
+    return False
+
+
+def _create_or_update_groups(account_groups):
+    """Creates the roles based on the groups provided."""
+    roles_id = []
+    for group in account_groups:
+        existing_role = current_datastore.find_role_by_id(group["id"])
+        if existing_role and existing_role.is_managed:
+            current_app.logger.exception(
+                f'Error while syncing roles: A managed role with id: ${group["id"]} already exists'
+            )
+            continue
+        existing_role_by_name = current_datastore.find_role(group["name"])
+        if existing_role_by_name and existing_role_by_name.is_managed:
+            current_app.logger.exception(
+                f'Error while syncing roles: A managed role with name: ${group["name"]} already exists'
+            )
+            continue
+        if not existing_role:
+            role = current_datastore.create_role(
+                id=group["id"],
+                name=group.get("name"),
+                description=group.get("description"),
+                is_managed=False,
+            )
+            roles_id.append(role.id)
+        elif existing_role and _role_needs_update(existing_role, group):
+            role_to_update = Role(
+                id=group["id"],
+                name=group.get("name"),
```

Comment:
As discussed, just double check that it can be `null`.

Reply:
Confirmed, name can be null 
![image](https://github.com/inveniosoftware/invenio-oauthclient/assets/15194802/1c11b190-46ba-4f6e-93c7-783d90a2cf3c)